server/webhook/issue.go

```diff
@@ -21,28 +33,71 @@ func (w *webhook) HandleIssue(event *gitlab.IssueEvent) ([]*HandleWebhook, error
 	}
 
 	if len(message) > 0 {
-		handlers := make([]*HandleWebhook, len(event.Assignees)+1)
+		toUsers := make([]string, len(event.Assignees)+1)
 		for index, assignee := range event.Assignees {
-			handlers[index] = &HandleWebhook{
-				Message: message,
-				To:      assignee.Username,
-				From:    senderGitlabUsername,
-			}
+			toUsers[index] = assignee.Username
 		}
-		handlers[len(handlers)-1] = &HandleWebhook{
+		toUsers[len(toUsers)-1] = authorGitlabUsername
+
+		handlers := []*HandleWebhook{{
 			Message: message,
-			To:      authorGitlabUsername,
+			ToUsers: toUsers,
 			From:    senderGitlabUsername,
-		}
+		}}
 
-		mentions := w.handleMention(mentionDetails{
+		if mention := w.handleMention(mentionDetails{
 			senderUsername:    senderGitlabUsername,
 			pathWithNamespace: event.Project.PathWithNamespace,
 			IID:               event.ObjectAttributes.IID,
 			URL:               event.ObjectAttributes.URL,
 			body:              event.ObjectAttributes.Description,
-		})
-		return cleanWebhookHandlers(append(handlers, mentions...)), nil
+		}); mention != nil {
+			handlers = append(handlers, mention)
+		}
+		return handlers, nil
 	}
 	return []*HandleWebhook{}, nil
 }
+
+func (w *webhook) handleChannelIssue(event *gitlab.IssueEvent) ([]*HandleWebhook, error) {
+	issue := event.ObjectAttributes
+	senderGitlabUsername := event.User.Username
+	repo := event.Project
+	res := []*HandleWebhook{}
+
+	message := ""
+
+	if issue.Action == "open" {
```

Comment:
Maybe use a `switch` with a `default` case?

Reply:
I can use a `switch` but not the `default` case because they are lot of possibilities for `issue.Action` and we don't want to treat all.